src/node_contextify.cc

```diff
@@ -463,18 +465,14 @@ class ContextifyContext {
       return;
 
     Local<Context> context = ctx->context();
-    Maybe<PropertyAttribute> maybe_prop_attr =
-        ctx->sandbox()->GetRealNamedPropertyAttributes(context, property);
-
-    if (maybe_prop_attr.IsNothing()) {
-      maybe_prop_attr =
-          ctx->global_proxy()->GetRealNamedPropertyAttributes(context,
-                                                              property);
-    }
+    MaybeLocal<Value> maybe_prop_desc =
+        ctx->sandbox()->GetOwnPropertyDescriptor(context, property);
 
-    if (maybe_prop_attr.IsJust()) {
-      PropertyAttribute prop_attr = maybe_prop_attr.FromJust();
-      args.GetReturnValue().Set(prop_attr);
+    if (!maybe_prop_desc.IsEmpty()) {
+      Local<Value> prop_desc = maybe_prop_desc.ToLocalChecked();
+      if (!prop_desc->IsUndefined()) {
```

Comment:
Is this check necessary?

Reply:
When the property does not exist on the sandbox, the property descriptor is undefined. In that case we don't intercept the call (so that `Object.getOwnPropertyDescriptor(this, 'Object')` can get it from the global object)